src/ape/api/providers.py

```diff
@@ -382,6 +382,14 @@ class ProviderAPI:
     request_header: str
     """A header to set on HTTP/RPC requests."""
 
+    def __getattr__(self, item):
+        # Check if trying to use a 'TestProvierAPI' method.
+        test_provider_api_members = [m for m in dir(TestProviderAPI) if m not in dir(ProviderAPI)]
+        if item in test_provider_api_members:
+            raise NotImplementedError(f"{self.name} is not a 'TestProviderAPI' implementation.")
+
+        raise NotImplementedError(f"ProviderAPI has no member '{item}'.")
```

Comment:
The only trick to `__getattr__` is that it *has* to raise `AttributeError` otherwise it gets into an infinite loop

Also, should probably call the `super` method
```suggestion
            raise AttributeError(f"{self.name} is not a 'TestProviderAPI' implementation.")

        raise AttributeError(f"ProviderAPI has no member '{item}'.")
        
        return super().__getattr__(item)
```

Handle the attribute error whenever you try to use a `TestProviderAPI` method, and just `raise NotImplementedError from attr_err`

Reply:
Perhaps a more simpler method is still this though: https://github.com/ApeWorX/ape/pull/402#discussion_r785619966

A bit more work to set up, as you'll have to maintain the methods in both places (where one is raising `NotImplementedError` and the other is `abstractmethod`) but perhaps you can filter the set of method names in `TestProviderAPI` that are `abstractmethod` and not in `ProviderAPI`, and add those as methods to `ProviderAPI` where they all just raise `NotImplementedError` instead.

Reply:
Yeah I want to avoid having to remember to add the methods to two separate spaces or else we will likely forget to do that in the future

Reply:
I will try to keep it dynamic without using the `__getattr__` approach